hedera-node/src/main/java/com/hedera/services/store/contracts/precompile/HTSPrecompiledContract.java

```diff
@@ -747,6 +753,11 @@ public long getMinimumFeeInTinybars(final Timestamp consensusTime) {
 		@Override
 		public TransactionBody.Builder body(final Bytes input, final UnaryOperator<byte[]> aliasResolver) {
 			mintOp = decoder.decodeMint(input);
+			try {
+				validateAndSetRevertReasonIfNeeded(mintOp.amount().compareTo(BigInteger.valueOf(Long.MAX_VALUE)) < 1, INVALID_TOKEN_MINT_AMOUNT);
+			} catch (InvalidTransactionException e) {
+				return null;
+			}
 			return syntheticTxnFactory.createMint(mintOp);
```

Comment:
```suggestion
			mintOp = decoder.decodeMint(input);
			if (mintOp.amount().compareTo(BigInteger.valueOf(Long.MAX_VALUE)) < 1) {
				messageFrame.setRevertReason(INVALID_MINT_AMOUNT_ERROR);
				return null;
			}
			return syntheticTxnFactory.createMint(mintOp);
```
I think this looks better and is easier to read. Also, the exception throwing/handling mechanism in java isn't the most efficient programming flow.
We can also define an instance variable for the error Bytes like we have for other error types (e.g. `ERROR_DECODING_INPUT_REVERT_REASON`):
	private static final Bytes INVALID_MINT_AMOUNT_ERROR = Bytes.of(INVALID_TOKEN_MINT_AMOUNT.name().getBytes());